extensions/wikia/TemplateClassification/services/UserTemplateClassificationService.class.php

```diff
@@ -35,12 +44,37 @@ class UserTemplateClassificationService extends TemplateClassificationService {
 	 * @throws \Swagger\Client\ApiException
 	 */
 	public function getType( $wikiId, $pageId ) {
-		$templateType = parent::getType( $wikiId, $pageId );
+		return $this->getMappedType( parent::getType( $wikiId, $pageId ) );
+	}
+
+	/**
+	 * Check if a given type is mapped as an infobox one.
+	 * @param string $type
+	 * @return bool
+	 */
+	public function isInfoboxType( $type ) {
+		return in_array( $type, self::$infoboxTypes );
+	}
+
+	protected function prepareTypes( $types ) {
+		$templateTypes = [];
+
+		foreach ( $types as $type ) {
+			$templateTypes[$type->getPageId()] = $this->getMappedType( $type->getType() );
+		}
+
+		return $templateTypes;
+	}
+
+	private function getMappedType( $type ) {
+		if ( $this->isInfoboxType( $type ) ) {
+			return self::TEMPLATE_INFOBOX;
+		}
 
-		if ( !in_array( $templateType, self::$templateTypes ) ) {
-			$templateType = self::TEMPLATE_UNCLASSIFIED;
+		if ( !in_array( $type, self::$templateTypes ) ) {
```

Comment:
Definitely not in the current setup. The last thing we want is to make the service, which has a higher "hierarchy", to use methods from a Special Page controller. We could move `isUserType` to the service itself, but let's not mix different scopes here. It does not bother me that much since the method itself is just this `in_array()` call.